### PR TITLE
done task 4

### PR DESCRIPTION
**Pull request by dasartini**
Finished the 4th please tell me if a need more error tests

### CODE REVIEW

File: MVC/app.js [hunk 1]
```diff
@@ -1,14 +1,27 @@
 const express = require('express')
 const app = express()
-
-
-// my different cb functions form controllers
-const {getTopics, beOnApi} = require('./controllers')
+const {getTopics, getApiEndPoints, getArticleById} = require('./controllers')
 
 app.use(express.json())
```
Comment (nc-saima-khanom) [inline MVC/app.js:5]: Has this line of code been driven by a test? Every code written shoul dbe driven by a test or part of a test but we haven't yet written a test that requires express.json() yet

File: MVC/controllers.js [hunk 2]
```diff
@@ -1,19 +1,30 @@
...
 
     })
 };
 
-const beOnApi = function(req, res, next){
-    console.log(endpoint)
-    return res.status(200).send({endpoint})  
-}
+const getApiEndPoints = function (req, res, next) {
+    return res.status(200).send({ endpoint })
```
Comment (nc-saima-khanom) [inline MVC/controllers.js:13]: as its an array, it makes sense to have this as 'endpoints' plural :)

File: MVC/controllers.js [hunk 3]
```diff
@@ -1,19 +1,30 @@
...
+};
+
+const getArticleById = function (req, res, next) {
+    const { article_id } = req.params
+   //console.log(article_id)
```
Comment (nc-saima-khanom) [inline MVC/controllers.js:18]: remove comments before pushing

File: MVC/controllers.js [hunk 4]
```diff
@@ -1,19 +1,30 @@
...
+    return fetchArticleId(article_id).then((article) => {
+        console.log({ article })
```
Comment (nc-saima-khanom) [inline MVC/controllers.js:20]: remove logs before pushing

File: MVC/models.js [hunk 5]
```diff
@@ -1,13 +1,22 @@
...
+    })
 };
 
+function fetchArticleId(article_id) {
+
+    return db.query(`
+SELECT * FROM articles WHERE article_id = $1;`, [article_id])
+        .then(({ rows }) => {
+            //console.log(rows.length)
+            if(rows.length ===0){  return Promise.reject({status: 400, message: "The id provided does not exist!"})}
```
Comment (nc-saima-khanom) [inline MVC/models.js:16]: 400 code is a bad request, what code should we send when an article isn't found?

File: __tests__/integration.test.js [hunk 6]
```diff
@@ -33,36 +33,81 @@ describe("/api/topics", () => {
...
+
+describe("/api/articles", () => {
+    test("GET:/api/articles/article:id returns the specified article", () => {
+        return request(app)
+            .get('/api/articles/6')
+            .expect(200)
+            .then(({ body }) => {
+                const { article } = body
+                expect(typeof article).toBe("object")
+                expect(article.hasOwnProperty('article_id')).toBe(true)
```
Comment (nc-saima-khanom) [inline __tests__/integration.test.js:81]: Another matcher that may be useful to look into: .toMatchObject()

File: __tests__/integration.test.js [hunk 7]
```diff
@@ -33,36 +33,81 @@ describe("/api/topics", () => {
...
+        return request(app)
+        .get('/api/articles/papa_pear')
+        .expect(400)
+        .then(({ body }) => {
+            const { message } = body;
+            expect(message).toBe("Bad request :(")
+        })
+    })
+
+    test.only("GET: Returns an error message when provided an higher id number",()=>{
```
Comment (nc-saima-khanom) [inline __tests__/integration.test.js:103]: remove .only from your tests before pushing otherwise we can't be sure that all of your tests pass!

Comment (nc-saima-khanom) [inline __tests__/integration.test.js:103]: this can be described as a valid but non-existent article id

File: endpoints.json [hunk 8]
```diff
@@ -25,5 +25,8 @@
         }
       ]
     }
+  },
```
Comment (nc-saima-khanom) [inline endpoints.json:28]: I'd like to see some more info on this endpoint - follow the format of the /api/articles example above for this

File: MVC/app.js (hunk 1 above)
Comment (dasartini) [reply MVC/app.js:5]: oh that is just something I always put as a default syntax, I just noticed as well none of the other changes I did like erased comments seems to be visible

File: MVC/controllers.js (hunk 2 above)
Comment (dasartini) [reply MVC/controllers.js:13]: I tried doing that so, and It breaks the test I dont know why

**Pull request merged**